scripts/run_evaluation.py

```diff
@@ -446,7 +450,20 @@ def export_experiment_info(
 
 if __name__ == "__main__":
     args = parse_and_validate_args()
-    tuned_model = TunedCausalLM.load(args.model, use_flash_attn=args.use_flash_attn)
+
+    base_model_name_or_path = args.base_model_name_or_path
+    if not base_model_name_or_path:
+        adapter_config_path = os.path.join(args.model, "adapter_config.json")
+        if os.path.exists(adapter_config_path):
+            with open(adapter_config_path, "r", encoding="utf-8") as config_file:
+                adapter_config = json.load(config_file)
+                base_model_name_or_path = adapter_config.get("base_model_name_or_path")
+
+    tuned_model = TunedCausalLM.load(
```

Comment:
If I'm reading this correctly, this code is missing the case that a user is running evaluation on a fine tuned model and there's no `adapter_config.json` file and they don't pass in any value for `base_model_name_or_path`. Then the model will fail to load, and an exception is raised, is that correct?

Reply:
That's a good point, the problem is that fine tuned models don't have an adapter_config.json nor is a base model needed to load the model. So this may be another bug in the run_inference script for fine tuned models that needs to be fixed. Let me try it out with fine tuned model...

Reply:
@aluu317 verified that both changes did not affect the fine tuning case. No change is needed for fine tuning because of [this line](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/scripts/run_inference.py#L182) in `TunedCausalLM.load()`. Since no adapter_config.json is found, this jumps below to the except block where the model is correctly loaded as a merged model.